magefiles/utils.go

```diff
@@ -150,3 +165,120 @@ func renderTemplate(destination, templatePath string, templateData interface{},
 
 	return nil
 }
+
+// helper function used for testing jvm-build-service PRs
+func overridePipelineBundle() error {
+	const buildTemplatesBundleKustomizeUrl = "https://raw.githubusercontent.com/redhat-appstudio/infra-deployments/main/components/build-templates/kustomization.yaml"
+	const bundleName = "hacbs_build_bundle"
+	var newReqprocessorImage = os.Getenv("JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE")
+
+	const imagePushRepo = "quay.io/redhat-appstudio-qe/test-images"
+	prSHA := openshiftJobSpec.Refs.Pulls[0].SHA
+	var newS2iJavaTaskRef, _ = name.ParseReference(fmt.Sprintf("%s:task-bundle-%s", imagePushRepo, prSHA))
+	var newJavaBuilderPipelineRef, _ = name.ParseReference(fmt.Sprintf("%s:pipeline-bundle-%s", imagePushRepo, prSHA))
+
+	var rawRegistryCreds []byte
+	var err error
+	var homeDir string
+
+	var newTaskYaml, newPipelineYaml []byte
+
+	registryCreds := os.Getenv("QUAY_TOKEN")
+	if rawRegistryCreds, err = base64.StdEncoding.DecodeString(registryCreds); err != nil {
+		return fmt.Errorf("unable to decode container registry credentials: %v", err)
+	}
+	if homeDir, err = homedir.Dir(); err != nil {
+		return fmt.Errorf("unable to locate home directory: %v", err)
+	}
+	if err := os.WriteFile(homeDir+"/.docker/config.json", rawRegistryCreds, 0644); err != nil {
+		return fmt.Errorf("failed to create a docker config file: %v", err)
+	}
+
+	keychain := authn.NewMultiKeychain(authn.DefaultKeychain)
+	authOption := remoteimg.WithAuthFromKeychain(keychain)
+
+	res, err := http.Get(buildTemplatesBundleKustomizeUrl)
+	if err != nil {
+		return fmt.Errorf("failed to get a build templates bundle: %v", err)
+	}
+	defer res.Body.Close()
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read the body response of a build templates bundle: %v", err)
+	}
+	re := regexp.MustCompile(fmt.Sprintf(`%s=(.*)\n`, bundleName))
+	matches := re.FindStringSubmatch(string(body))
+	if len(matches) < 2 {
+		log.Printf("couldn't find a default build bundle in kustomize %s", buildTemplatesBundleKustomizeUrl)
+	}
+	defaultBundle := matches[1]
+	klog.Infof("default pipeline bundle ref: %s", defaultBundle)
+
+	resolver := oci.NewResolver(defaultBundle, authn.DefaultKeychain)
```

Comment:
nit: it feels like this set of code around the oci resolver init and tekton object retrieval could be duplicated into a single function and maybe interrogates the runtime.Object type using `GetObjectKind()` to then determine what type of tekton object it needs to return for you to manipulate?